services/streammanager/kafka/kafkamanager.go

```diff
@@ -394,9 +401,48 @@ func prepareMessage(topic, key string, message []byte, timestamp time.Time) clie
 	}
 }
 
-func prepareBatchOfMessages(topic string, batch []map[string]interface{}, timestamp time.Time) (
+// This function is used to create the error string of idividual schema
+func createErrString(err error, index int) string {
+	return fmt.Sprintf(" Schema[%d]: ", index+1) + err.Error() + "."
+}
+
+// This function is used to serialize the binary data according to the AVROSchema.
+// It iterate over the schemas provided by the customer and try to serialize the data.
+// If it ables to serialize the data then it returns the converted data otherwise it throws error.
+// We are using linkedin goavro library to serialize the data. For ref: https://github.com/linkedin/goavro
+
+func serialize(value []byte, AVROSchema []Schema) ([]byte, error) {
+	var errorStr string
+	for i := 0; i < len(AVROSchema); i++ {
+		codec, err := goavro.NewCodec(AVROSchema[i].AVROSchema)
```

Comment:
Okay, we can move the codec declaration to the `NewProducer` section.

Reply:
Global variables are not a good practice most of the time so unless you have a very good reason please try to avoid them. What do you think about passing it around via `producerImpl`?

Reply:
Yes I have passed it through `producerImpl`